src/uShip.Logging/LogBuilders/PropertiesDictionaryExtensions.cs

```diff
@@ -11,14 +14,49 @@ private static string Sanitize(string input)
             return input.IfNotNull(x => x.SanitizeSensitiveInfo().RemoveViewState());
         }
 
+        public static IDictionary NewSanitizedDictionary(this IDictionary data)
+        {
+            if (data == null)
+            {
+                return null;
+            }
+
+            IDictionary sanitizedData = new Dictionary<String, object>();
+
+            foreach (var key in data.Keys)
+            {
+                if (data[key] == null)
+                {
+                    continue;
+                }
+
+                if (data[key] is string && LogDataSanitizeExtensions.SensitiveFieldNames.Contains(key))
```

Comment:
I dont understand.  This recursively goes down and makes the change.  Do you have another suggested solution?